src/registry/RegistryFactory.php

```diff
@@ -10,64 +10,125 @@
-        'npm'   => NpmRegistry::class,
+    public $registryMap = [
+        //'bower' => BowerRegistry::class,
+        //'npm'   => NpmRegistry::class,
     ];
 
     /**
-     * @var AbstractAssetsRepository[]
+     * @var string|IOInterface
      */
-    public static $registries = [];
+    public $io = [
+        'class' => YiiLogIO::class,
+    ];
 
     /**
-     * @param string $type
-     * @param RepositoryManager $rm
-     * @return AbstractAssetsRepository|BowerRegistry|NpmRegistry
+     * @var ComposerConfig 
      */
-    public static function getRegistry($type, $rm)
-    {
-        if (!isset(static::$registries[$type])) {
-            static::$registries[$type] = static::buildRegistry($type, $rm);
-        }
+    public $composerConfig;
 
-        return static::$registries[$type];
-    }
+    /**
+     * @var RepositoryManager
+     */
+    public $repositoryManager;
+
+    /**
+     * @var AssetConfig 
+     */
+    public $assetConfig;
+
+    /**
+     * @var RootPackage 
+     */
+    public $rootPackage;
+
+    /**
+     * @var InstallationManager 
+     */
+    public $installationManager;
+
+    /**
+     * @var VcsPackageFilter 
+     */
+    public $packageFilter;
 
     /**
-     * @param string $type
-     * @param RepositoryManager $rm
-     * @return mixed
+     * @var AssetRepositoryManager 
      */
-    protected static function buildRegistry($type, $rm)
+    public $assetRepositoryManager;
+
+    public function init()
     {
-        $class = static::$classes[$type];
-        $rm->setRepositoryClass($type, $class);
+        parent::init();
+
+        $this->io = Instance::ensure($this->io, IOInterface::class);
+
+        $this->composerConfig = Factory::createConfig($this->io);
+
+        $this->io->loadConfiguration($this->composerConfig);
+
+        $this->repositoryManager = RepositoryFactory::manager($this->io, $this->composerConfig);
+
+        $arrayConfig = [];
+        if ($this->composerConfig->has('fxp-asset')) {
+            $arrayConfig = $this->composerConfig->get('fxp-asset');
+        }
+
+        $this->assetConfig = new AssetConfig($arrayConfig);
+
+        $this->rootPackage = new RootPackage('asset-packagist', '0.0.0.0', '0.0.0');
+
+        $this->installationManager = new InstallationManager();
 
-        $config = [
-            'asset-repository-manager' => self::createAssetRepositoryManager($rm),
-            'asset-options'      => [],
-        ];
+        $this->packageFilter = new VcsPackageFilter($this->assetConfig, $this->rootPackage, $this->installationManager);
 
-        return $rm->createRepository($type, $config);
+        $this->assetRepositoryManager = new AssetRepositoryManager($this->io, $this->repositoryManager, $this->assetConfig, $this->packageFilter);
+
+        //This not allow override class
+        AssetPlugin::addRegistryRepositories($this->assetRepositoryManager, $this->packageFilter, $this->assetConfig);
+        //
+        //This allow override class
```

Comment:
It would be nice you to give a general description of RegistryFactory refactoring, please.